scipy/sparse/matrix_io.py

```diff
@@ -0,0 +1,149 @@
+import numpy as np
+import scipy.sparse
+
+
+def save(file, matrix, compressed=True):
+    """ Save a sparse matrix.
+    
+    Save the sparse matrix into a file in ``.npz`` format.
+    If compressed, the resulting file is compressed.
+
+    Parameters
+    ----------
+    file : str or file
+        Either the file name (string) or an open file (file-like object)
+        where the data will be saved. If file is a string, the ``.npz``
+        extension will be appended to the file name if it is not already
+        there.
+    matrix: spmatrix (format: ``csc``, ``csr`` or ``bsr``)
+        The sparse matrix to save.
+    compressed : bool, optional
+        Allow compressing the file. Default: True
+
+    Returns
+    -------
+    None
+
+    See Also
+    --------
+    numpy.savez: Save several arrays into a ``.npz`` archive.
+    numpy.savez_compressed : Save several arrays into a compressed ``.npz`` archive.
+
+    Notes
+    -----
+    The ``.npz`` file format is a zipped archive of files named after the
+    variables they contain. The archive containts the three ``.np`` files
+    ``data``, ``indices`` and ``indptr`` containing the corresponding values
+    of the sparse matrix.
+
+    Examples
+    --------
+    Store sparse matrix to disk, and load it again:
+
+    >>> import scipy.sparse
+    >>> sparse_matrix = scipy.sparse.csc_matrix(np.array([[0, 0, 3], [4, 0, 0]]))
+    >>> sparse_matrix
+    <2x3 sparse matrix of type '<type 'numpy.int64'>'
+       with 2 stored elements in Compressed Sparse Column format>
+    >>> sparse_matrix.todense()
+    matrix([[0, 0, 3],
+            [4, 0, 0]], dtype=int64)
+
+    >>> scipy.sparse.save('/tmp/sparse_matrix.npz', sparse_matrix)
+    >>> sparse_matrix = scipy.sparse.load('/tmp/sparse_matrix.npz')
+
+    >>> sparse_matrix
+    <2x3 sparse matrix of type '<type 'numpy.int64'>'
+       with 2 stored elements in Compressed Sparse Column format>
+    >>> sparse_matrix.todense()
+    matrix([[0, 0, 3],
+            [4, 0, 0]], dtype=int64)
+    """
+
+    if matrix.format in ('csc', 'csr', 'bsr'):
+        arrays_dict = {'format': np.array(matrix.format), 'shape': np.array(matrix.shape), 'data': matrix.data, 'indices': matrix.indices, 'indptr': matrix.indptr}
+    elif matrix.format == 'dia':
+        arrays_dict = {'format': np.array(matrix.format), 'shape': np.array(matrix.shape), 'data': matrix.data, 'offsets': matrix.offsets}
+    elif matrix.format == 'coo':
+        arrays_dict = {'format': np.array(matrix.format), 'shape': np.array(matrix.shape), 'data': matrix.data, 'row': matrix.row, 'col': matrix.col}
```

Comment:
There's no need to convert everything to arrays here, `savez` will automatically do that for you.

Also, it might be nice to pull out the common items:

```python
arrays_dict = dict(format=matrix.format, shape=matrix.shape, data=matrix.data)
if matrix.format in ('csc', 'csr', 'bsr'):
    arrays_dict.update(indices=matrix.indices, indptr=matrix.indptr)
elif matrix.format == 'dia':
    arrays_dict['offsets'] = matrix.offsets
elif matrix.format == 'coo':
    arrays_dict.update(row=matrix.row, col=matrix.col)
```